plugins/sentry/src/components/Router.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { Routes, Route } from 'react-router';
+import { WarningPanel } from '@backstage/core';
+import { SentryPluginWidget } from './SentryPluginWidget/SentryPluginWidget';
+
+const SENTRY_ANNOTATION = 'sentry.io/project-id';
+
+const isPluginApplicableToEntity = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.[SENTRY_ANNOTATION]) &&
+  entity.metadata.annotations?.[SENTRY_ANNOTATION] !== '';
```

Comment:
The Boolean actually covers the empty string (an empty string is falsy) - you should skip everything after `&&`

Reply:
however, at the first line of the router, you could just do

```ts
const projectId = entity.metadata.annotations?.[SENTRY_ANNOTATION]);
if (!projectId) { return the warning panel }
// here we know that a project ID was given and was non-empty
// so continue just rendering
...
<SentryPluginWidget sentryProjectId={projectId} statsFor="24h" />
```

Reply:
Addressed this and the other comments. Thanks for your time. 🙌